gnomad/utils/annotations.py

```diff
@@ -1188,3 +1190,132 @@ def hemi_expr(
+                    i.get(k, [hl.missing(hl.tint32)]).extend(
+                        j.get(k, [hl.missing(hl.tint32)])
+                    ),
+                )
+            )
+        ),
+        fmeta[0],
+        fmeta[1:],
+    )
+
+    # Create a list of tuples from the dictionary, sorted by the list of indices for
+    # each aggregation group.
+    fmeta = hl.sorted(fmeta.items(), key=lambda f: f[1])
+
+    # Create a list of the aggregation groups, maintaining the sorted order
+    new_freq_meta = fmeta.map(lambda x: x[0])
+
+    # Create array for each aggregation group of arrays containing the group's freq
+    # values from each freq array
+    freq_meta_idx = fmeta.map(lambda x: hl.zip(farrays, x[1]).map(lambda i: i[0][i[1]]))
+
+    def _sum_or_diff_fields(
+        field_1_expr: str, field_2_expr: str
+    ) -> hl.expr.Int32Expression:
+        """
+        Sum or subtract fields in call statistics struct.
+
+        :param field_1_expr: First field to sum or diff.
+        :param field_2_expr: Second field to sum or diff.
+        :return: Merged field value.
+        """
+        return hl.if_else(
+            operation == "sum",
+            hl.or_else(field_1_expr, 0) + hl.or_else(field_2_expr, 0),
+            hl.or_else(field_1_expr, 0) - hl.or_else(field_2_expr, 0),
+        )
+
+    # Iterate through the groups and their freq lists to merge callstats
+    new_freq = freq_meta_idx.map(
+        lambda x: hl.bind(
+            lambda y: y.annotate(AF=hl.if_else(y.AN > 0, y.AC / y.AN, 0)),
+            hl.fold(
+                lambda i, j: hl.struct(
+                    AC=_sum_or_diff_fields(i.AC, j.AC),
+                    AN=_sum_or_diff_fields(i.AN, j.AN),
+                    homozygote_count=_sum_or_diff_fields(
+                        i.homozygote_count, j.homozygote_count
+                    ),
+                ),
+                x[0].select("AC", "AN", "homozygote_count"),
+                x[1:],
+            ),
+        )
+    )
+    # Check and see if any annotation within the merged array is negative. If so,
+    # raise an error if set_negatives_to_zero is False or set the value to 0 if
+    # set_negatives_to_zero is True.
+    if operation == "diff":
+        negative_value_error_msg = (
+            "Negative values found in merged frequency array. Review data or set"
+            " `set_negatives_to_zero` to True to set negative values to 0."
+        )
+        new_freq = new_freq.map(
+            lambda x: x.annotate(
+                AC=hl.case()
+                .when(set_negatives_to_zero, hl.max(x.AC, 0))
+                .or_error(negative_value_error_msg),
+                AN=hl.case()
+                .when(set_negatives_to_zero, hl.max(x.AN, 0))
+                .or_error(negative_value_error_msg),
+                homozygote_count=hl.case()
+                .when(set_negatives_to_zero, hl.max(x.homozygote_count, 0))
+                .or_error(negative_value_error_msg),
+            )
```

Comment:
**{
                    ann: hl.case()
                           .when(set_negatives_to_zero, hl.max(x[ann], 0))
                           .or_error(negative_value_error_msg)
                    for ann in callstat_ann
                }